Tella/Scenes/Uwazi/ViewModel/AddTemplateVM.swift

```diff
@@ -68,9 +71,13 @@ class AddTemplateViewModel: ObservableObject {
     }
     
     func downloadTemplate(template: CollectedTemplate) {
-        var template = template
-        self.downloadTemplate(template: &template)
-        self.templateItemsViewModel.first(where: {template.templateId == $0.id})?.isDownloaded = true
+        self.isLoading = true
+        entityFetcher?.fetchRelationshipEntities(template: template) { relationships in
+            template.relationships = relationships
+            self.saveTemplate(template: template)
+            self.isLoading = false
+            self.templateItemsViewModel.first(where: {template.templateId == $0.id})?.isDownloaded = true
+        }
```

Comment:
The current code doesn't handle potential errors during relationship fetching or template saving ... let's improve this ...

Reply:
you are right! I just refactor both fetchRelationship and `addUwaziTemplate` to return a Result type to catch any error that happens during the process